### DISCUSSION TITLE
如何实现自定义数据集+finetune完成情感分析呢？

### DISCUSSION THREAD

**Discussion by stronger123-boa** [PaddlePaddle/PaddleNLP#374] [Q&A] [answered]
使用自定义数据集，构建data_loader.

**Comment by stronger123-boa** [comment]
使用自定义数据集，构建data_loader《使用PaddleNLP语义预训练模型ERNIE优化情感分析》的create_dataloader。报错：没有.map()功能。改成用《快来选一顿好吃的年夜饭》的create_dataloader，报错：SystemError: (Fatal) Blocking queue is killed because the data reader raises an exception.  请问有什么好的案例推荐吗

**Comment by Steffy-zxf** [comment, marked as answer]
你好！请升级PaddleNLP为最新版本
```
pip install --upgrade paddlenlp
```

自定义数据集参考文档：https://github.com/PaddlePaddle/PaddleNLP/blob/develop/docs/data_prepare/dataset_self_defined.rst

另外，方便贴下错误信息以及代码吗？
提供以上信息可以帮助分析问题。
